File: src/BjjEire.Api/Constants/FeatureFlags.cs
// Copyright (c) BjjEire. All rights reserved.
// Licensed under the MIT License.

namespace BjjEire.Api.Constants;

public static class FeatureFlags
{
    public const string BjjEvents = "BjjEvents";
    public const string Gyms = "Gyms";
    public const string Competitions = "Competitions";
    public const string Stores = "Stores";

    public static readonly IReadOnlyList<string> FrontendFlags =
    [
        BjjEvents,
        Gyms,
        Competitions,
        Stores,
    ];
}

File: src/bjjeire-app/src/testing/msw/handlers/feature-flags.ts
import { http, HttpResponse } from 'msw'
import { API_ROUTES } from '@/config/api-routes'
import type { FeatureFlagsMap } from '@/features/feature-flags'

export const defaultFeatureFlags: FeatureFlagsMap = {
  BjjEvents: true,
  Gyms: true,
  Competitions: true,
  Stores: true,
}

export const featureFlagsHandlers = [
  http.get(`*${API_ROUTES.featureFlags}`, () =>
    HttpResponse.json(defaultFeatureFlags)
  ),
]
